controller/array_action/array_mediator_svc.py

```diff
@@ -761,15 +762,14 @@ def get_first_free_lun(self, host_name):
         # can be mapped to a single host. (Note that some hosts such as linux
         # do not support more than 255 or 511 mappings today irrespective of
         # our constraint).
-        for candidate in range(self.MIN_LUN_NUMBER, self.MAX_LUN_NUMBER + 1):
-            if str(candidate) not in luns_in_use:
-                logger.debug("First available LUN number for {0} is "
-                             "{1}".format(host_name, str(candidate)))
-                lun = str(candidate)
-                break
+        lun_range = list(range(self.MIN_LUN_NUMBER, self.MAX_LUN_NUMBER + 1))
+        lun_range_str = [str(lun) for lun in lun_range]
```

Comment:
could we postpone the casting to the last operation of this function (as in current develop)?

Reply:
oh, I see current develop has multiple casting, and that we get strings from pysvc.
resolving

Reply:
we need it for the [next line](https://github.com/IBM/ibm-block-csi-driver/blob/8936936bca13232d29147b20fd9fea3727133048/controller/array_action/array_mediator_svc.py#L767), `luns_in_use` is a string list so `lun_range` should be too..